packages/block-editor/src/store/selectors.js

```diff
@@ -2891,6 +2887,26 @@ export function __unstableIsWithinBlockOverlay( state, clientId ) {
 export const getBlockEditingMode = createRegistrySelector(
 	( select ) =>
 		( state, clientId = '' ) => {
+			// In zoom-out mode, override the behavior set by
+			// __unstableSetBlockEditingMode to only allow editing the top-level
+			// sections.
+			const editorMode = __unstableGetEditorMode( state );
+			if ( editorMode === 'zoom-out' ) {
+				if ( clientId === '' /* ROOT_CONTAINER_CLIENT_ID */ ) {
+					return 'disabled';
+				}
+
+				const { sectionRootClientId } = unlock( getSettings( state ) );
+				if ( clientId === sectionRootClientId ) {
+					return 'contentOnly';
+				}
+
+				const sectionsClientIds = getBlockOrder( state );
+				if ( sectionsClientIds.includes( clientId ) ) {
+					return 'default';
```

Comment:
Should this return "default" or instead fallback to the remaining logic of the selector?